x-pack/plugins/upgrade_assistant/public/application/components/kibana_deprecations/steps_modal.tsx

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FunctionComponent } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import {
+  EuiText,
+  EuiSteps,
+  EuiSpacer,
+  EuiButton,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiTitle,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+} from '@elastic/eui';
+
+export interface StepsModalContent {
+  domainId: string;
+  steps: string[];
+  documentationUrl?: string;
+}
+
+interface Props {
+  closeModal: () => void;
+  modalContent: StepsModalContent;
+}
+
+const i18nTexts = {
+  getModalTitle: (domainId: string) =>
+    i18n.translate('xpack.upgradeAssistant.kibanaDeprecations.stepsModal.modalTitle', {
+      defaultMessage: "Fix '{domainId}'",
+      values: {
+        domainId,
+      },
+    }),
+  getStepTitle: (step: number) =>
+    i18n.translate('xpack.upgradeAssistant.kibanaDeprecations.stepsModal.stepTitle', {
+      defaultMessage: 'Step {step}',
+      values: {
+        step,
+      },
+    }),
+  modalDescription: i18n.translate(
+    'xpack.upgradeAssistant.kibanaDeprecations.stepsModal.modalDescription',
+    {
+      defaultMessage: 'Follow the steps below to address this deprecation.',
```

Comment:
I'd use resolve instead of introducing another word for fix. Again, it would be better if this could be specific, rather than a generic intro. If it has to be generic, "To resolve this deprecation:" or, maybe just omit the intro entirely? It doesn't really add anything.